pkgs/development/tools/click/default.nix

```
@@ -0,0 +1,90 @@
+{ lib
+, fetchFromGitLab
+, buildPythonPackage
+, autoconf
+, debian
+, automake
+, perl
+, libtool
+, vala
+, pkg-config
+, libgee
+, json-glib
+, properties-cpp
+, gobject-introspection
+, getopt
+, setuptools
+, pygobject3
+, wrapGAppsHook
+}:
+
+buildPythonPackage {
+  pname = "click";
+  version = "22-02-2023";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/click";
+    rev = "aaf2735e8e6cbeaf2e429c70136733513a81718a";
+    sha256 = "sha256-pNu995/w3tbz15QQVdVYBnWnAoZmqWj1DN/5PZZ0iZw=";
+  };
+
+
+  configurePhase = ''
+    ./autogen.sh
+    ./configure --prefix=$out \
+        --with-systemdsystemunitdir=/lib/systemd/system \
+        --with-systemduserunitdir=/usr/lib/systemd/user
+  '';
+
+  buildPhase = ''
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r {bin,click_package,lib} $out
+  '';
+
+  preFixup = ''
+    makeWrapperArgs+=(--prefix GI_TYPELIB_PATH : "$out/lib/click")
+  '';
+
+
+  format = "other";
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    perl
+    libtool
+    pkg-config
+    gobject-introspection.dev
+    vala
+    getopt
+    setuptools
+    wrapGAppsHook
+  ];
+
+  doCheck = false;
+
+  patches = [
+    ./dbus-test-runner.patch
+  ];
+
+  propagatedBuildInputs = [
+    debian
+    libgee
+    json-glib.dev
+    properties-cpp
+    pygobject3
+  ];
+
+
+  meta = {
+    description = "A tool to build click packages. Mainly used for Ubuntu Touch.";
+    homepage = "https://gitlab.com/ubports/development/core/click";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ ilyakooo0 ];
+  };
```

Missing `meta.platform`: https://nixos.org/manual/nixpkgs/stable/#var-meta-platforms

> The list of Nix platform types on which the package is supported. Hydra builds packages according to the platform specified. If no platform is specified, the package does not have prebuilt binaries.

```suggestion
    platforms = lib.platforms.linux;
  };
```